liblangutil/SourceReferenceExtractor.h

```diff
@@ -31,28 +31,27 @@ namespace langutil
 
 struct LineColumn
 {
-	int line;
-	int column;
+	int line {-1};
+	int column {-1};
 
+	LineColumn() = default;
 	LineColumn(std::tuple<int, int> const& _t): line{std::get<0>(_t)}, column{std::get<1>(_t)} {}
-	LineColumn(int _line, int _column): line{_line}, column{_column} {}
-	LineColumn(): line{-1}, column{-1} {}
 };
 
 struct SourceReference
 {
-	std::string message;    ///< A message that relates to this source reference (such as a warning or an error message).
-	std::string sourceName; ///< Underlying source name (for example the filename).
-	LineColumn position;    ///< Actual (error) position this source reference is surrounding.
-	bool multiline;         ///< Indicates whether the actual SourceReference is truncated to one line.
-	std::string text;       ///< Extracted source code text (potentially truncated if multiline or too long).
-	int startColumn;        ///< Highlighting range-start of text field.
-	int endColumn;          ///< Highlighting range-end of text field.
+	std::string message;       ///< A message that relates to this source reference (such as a warning or an error message).
+	std::string sourceName {}; ///< Underlying source name (for example the filename).
```

Comment:
@chriseth ths initializer can't be fully omitted. It serves like a default value if none is given when constructing this object.

Reply:
@christianparpart I'm still pretty sure, you're wrong about that - it will default construct even without this default value - just the same as ``message`` one line above, where you *did* remove the member-initialization already :-).
Non-PODs that have a default constructor will always be default constructed using their default-constructor, regardless of whether they have an empty member-initializer or not.

Reply:
@ekpyron the way the code was, with the above proposal applied, you run into a compile error (at least on my compiler, GCC 8.1), so I empirically fixed that until GCC was happy. Now, we decided to go with another approach anyways.

Reply:
What kind of error? The code should be exactly equivalent, so this would rather indicate a bug in GCC 8.1 then... In any case, if it's now fine, it doesn't matter too much.

Reply:
@chriseth I highly doubt that and in fact it doesn't, if I try it. The only constructors that the compiler ever generates implicitly are default, copy and move constructors. The only thing that would work is aggregate initialization but for that I can skip values anyways, regardless of whether they are member-initialized or not.